sklearn/metrics/cluster/_expected_mutual_info_fast.pyx

```diff
@@ -50,7 +50,8 @@ def expected_mutual_information(contingency, int n_samples):
     gln_N = gammaln(N + 1)
     gln_nij = gammaln(nijs + 1)
     # start and end values for nij terms for each summation.
-    start = np.array([[v - N + w for w in b] for v in a], dtype='int')
+    start = np.array(np.meshgrid(a,b),dtype='int').T
+    start = np.array([start[i].sum(axis=1)-N for i in range(len(start))])
```

Comment:
I'd be open to suggestions on how to do these operations using numpy in more elegant way

Reply:
The issue that I can spot here is that we are creating a huge matrix in memory.

Reply:
the size of matrix will be double than previous method. 

Precisely, if `a` has `M` elements and `b` has `N` elements, then

- by the old method you will have M*N elements
- by the new method you have 2*M\*N elements
```python
>>> a = np.array([1,2])
>>> b = np.array([3,4,5])
>>> np.meshgrid(a,b)
[array([[1, 2],
       [1, 2],
       [1, 2]]), 
array([[3, 3],
       [4, 4],
       [5, 5]])]
```

So although the difference exists, I don't think it should be the deciding factor.